services/pdf.py

```
@@ -0,0 +1,47 @@
+
+    # Main logic for processing pdf
+    @timer
+    def pdf_to_chunks(self, pdf_url):
+        data = self.get_pdf_from_url(pdf_url)
+        text = extract_text(data)
+        return self.chunk_text(text)
+
+    # Fetches a PDF from a given URL and return it as a BytesIO object
+    def get_pdf_from_url(self, url: str) -> BytesIO:
+        # Fetch content from the URL
+        response = requests.get(url)
```

Ruff (0.13.3)

Probable use of `requests` call without timeout

(S113)